src/flow/restricted_flow.jl

```diff
@@ -0,0 +1,93 @@
+"""
+Restricts a (capacity) matrix to a given value. That is, for any vertices u and v, a capacity function c, and a restriction value r, the new resctricted capacity function c' is given by c'(u,v) = min(c(u,v), r).
+
+Requires arguments:
+
+- capacity_matrix::AbstractArray{T,2}     # input capacity matrix
+- restriction::T                          # restriction on the capacities
+"""
+
+function restrict_capacity_matrix{T<:Number}(
+    capacity_matrix::AbstractArray{T,2},   # edge flow capacities
+    restriction::T                         # restriction over the capacities
+    )
+    restricted_matrix = copy(capacity_matrix)
+    return inner_restrict_capacity_matrix!(restricted_matrix, restriction)
+end
+
+# Inner method for restrict_capacity_matrix
+
+function inner_restrict_capacity_matrix!{T<:Number}(
+    restricted_matrix::AbstractArray{T,2},   # edge flow capacities
+    restriction::T                           # restriction over the capacities
+    )
+    for i in eachindex(restricted_matrix)
+      if restriction < restricted_matrix[i]
+        restricted_matrix[i] = restriction
+      end
+    end
```

Comment:
Me neither. Good to know. Do we still need to separate the two functions in this case? I couldn't find the file with the implementation of min (for arrays) in JuliaLang.

If not, I think @CarloLucibello solution is definitely the best. 
> an alternative way would be to give to maximum_flow a new optional argument, e.g. restriction=zero(T), instead of defining a new function restricted_maximum_flow

Anyway, I will just compress (L24-28) for now.